utils/project/path.go

```diff
@@ -74,17 +74,37 @@ func PackageForPath(directory string) (string, error) {
 }
 
 // Subdirectories walkthroughs all subdirectories. The results contains itself.
-// If a path is non-existent or not in GOPATH, the path will be ignored.
+// The path will be ignored if:
+// 1. The path is non-existent
+// 2. Go module is disabled and the path is not in GOPATH
+// 3. Go modules is enabled and the path doesn't belong to a go module
 func Subdirectories(vendor bool, paths ...string) []string {
 	walked := map[string]bool{}
 	goDir := map[string]bool{}
 	for _, path := range paths {
-		_, absPath, err := GoPath(path)
+		var (
+			absPath string
+			err     error
+		)
+		useModule := os.Getenv("GO111MODULE")
+		if useModule == "off" {
+			_, absPath, err = GoPath(path)
+		} else {
+			root := findModuleRoot(path)
+			if len(root) > 0 {
+				absPath, err = filepath.Abs(path)
+			} else {
+				if useModule == "on" {
+					continue
+				}
+				_, absPath, err = GoPath(path)
+			}
+		}
 		if err != nil {
 			// Ignore the err and go next.
 			continue
 		}
-		err = filepath.Walk(absPath, func(path string, info os.FileInfo, err error) error {
+		_ = filepath.Walk(absPath, func(path string, info os.FileInfo, err error) error {
```

Comment:
Why don't we handle this error here?

Reply:
I am not sure, it is not handled now:
https://github.com/caicloud/nirvana/blob/2fb93f51598f353e72bbbd313a5ecb8d77388828/utils/project/path.go#L107

how should we handle this error properly?

Reply:
How about panic here?

Reply:
I'm not so sure too, but I think we can keep ignoring it.

Reply:
OK SGTM